instrumentation/jetty-httpclient/jetty-httpclient-9.2/library/src/main/java/io/opentelemetry/instrumentation/jetty/httpclient/v9_2/internal/JettyClientHttpAttributesExtractor.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.jetty.httpclient.v9_2.internal;
+
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpAttributesExtractor;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpVersion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class JettyClientHttpAttributesExtractor extends HttpAttributesExtractor<Request, Response> {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(JettyClientHttpAttributesExtractor.class);
+
+  @Override
+  @Nullable
+  protected String method(Request request) {
+    return request.getMethod();
+  }
+
+  @Override
+  @Nullable
+  protected String url(Request request) {
+    return request.getURI().toString();
+  }
+
+  @Override
+  @Nullable
+  protected String target(Request request) {
+    String queryString = request.getQuery();
+    return queryString != null ? request.getPath() + "?" + queryString : request.getPath();
+  }
+
+  @Override
+  @Nullable
+  protected String host(Request request) {
+    return request != null ? request.getHost() : null;
+  }
+
+  @Override
+  @Nullable
+  protected String route(Request request) {
+    return null;
+  }
+
+  @Override
+  @Nullable
+  protected String scheme(Request request) {
+    return request != null ? request.getScheme() : null;
+  }
+
+  @Override
+  @Nullable
+  protected String userAgent(Request request) {
+    HttpField agentField = request.getHeaders().getField(HttpHeader.USER_AGENT);
+    return agentField != null ? agentField.getValue() : null;
+  }
+
+  @Override
+  @Nullable
+  protected Long requestContentLength(Request request, @Nullable Response response) {
+    Long reqContentLength = null;
+
+    if (request != null) {
+      HttpField requestContentLengthField =
+          request.getHeaders().getField(HttpHeader.CONTENT_LENGTH);
+      reqContentLength = getLongFromJettyHttpField(requestContentLengthField);
+    }
+
+    return reqContentLength;
+  }
+
+  @Override
+  @Nullable
+  protected Long requestContentLengthUncompressed(Request request, @Nullable Response response) {
+    return null;
+  }
+
+  @Override
+  @Nullable
+  protected String flavor(Request request, @Nullable Response response) {
+
+    if (response == null) {
+      return "1.1";
+    }
+
+    HttpVersion httpVersion = response.getVersion();
+    httpVersion = (httpVersion != null) ? httpVersion : HttpVersion.HTTP_1_1;
+    switch (httpVersion) {
+      case HTTP_0_9:
```

Comment:
Let's use `HttpFlavorValues` enum for the return value (and go ahead and return 1.0 for 0.9 since we don't have a spec definition for 0.9 and I guess no one uses it in the wild

Reply:
Good idea, done